### PR TITLE
Fixed CryptoRecordsSanityCheckSuite

### LINKED ISSUE #9501: Fix CryptoRecordsSanityCheckSuite
- [ ] `insufficientAccountBalanceRecordSanityChecks()`
- [ ] `invalidPayerSigCryptoTransferRecordSanityChecks()`

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Created an issue #9550 as we discussed in our modularization call today.

**Related issue(s)**:

Fixes #9501

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java [hunk 1]
```diff
@@ -110,11 +111,15 @@ public Account getPayerAccount(@NonNull final ReadableStoreFactory storeFactory,
      * @param txInfo the {@link TransactionInfo} to use during the check
      * @param account the {@link Account} with the balance to check
      * @param fees the fees to use for the check
+     * @param ingestCheck if true, the check is being performed during an ingest workflow. This is to match mono and pass HapiTest. Should reconsider later.ref #9550
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java:114]: This isn't a very visible place to put the issue reference. I'd suggest something above the method param:
```java
            @NonNull final Account account,
            @NonNull final Fees fees,
            // This is to match mono and pass HapiTest. Should reconsider later.
            // FUTURE ('#9550')
            boolean ingestCheck)
```
Also, `final boolean ingestCheck`

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java [hunk 2]
```diff
@@ -128,13 +133,21 @@ public void checkSolvency(
             throw new InsufficientNetworkFeeException(INSUFFICIENT_TX_FEE, totalFee);
         }
         if (availableBalance < fees.networkFee()) {
-            throw new InsufficientNetworkFeeException(INSUFFICIENT_PAYER_BALANCE, totalFee);
+            if (ingestCheck) { // throw different exception for ingest
+                throw new InsufficientBalanceException(INSUFFICIENT_PAYER_BALANCE, totalFee);
+            } else {
+                throw new InsufficientNonFeeDebitsException(INSUFFICIENT_ACCOUNT_BALANCE, totalFee);
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java:139]: Yick!  Definitely want to remove this false exception in the future.  Labeling insufficient funds to pay a _fee_ as `InsufficientNonFeeDebits` is just bad...

**Review by jsync-swirlds — DISMISSED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java (hunk 1 above)
Comment (jsync-swirlds) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java:114]: I've been recommending a modified annotation syntax, to make it easier to find with text search tools without getting false positives.
`// @future('9550')` or `// @todo('9550')`

Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java]: Updated with the change suggested.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java]: Updated with the change suggested.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java [hunk 3]
```diff
@@ -128,13 +135,21 @@ public void checkSolvency(
             throw new InsufficientNetworkFeeException(INSUFFICIENT_TX_FEE, totalFee);
         }
         if (availableBalance < fees.networkFee()) {
-            throw new InsufficientNetworkFeeException(INSUFFICIENT_PAYER_BALANCE, totalFee);
+            if (ingestCheck) { // throw different exception for ingest
+                throw new InsufficientNetworkFeeException(INSUFFICIENT_PAYER_BALANCE, totalFee);
+            } else {
+                throw new InsufficientNetworkFeeException(INSUFFICIENT_ACCOUNT_BALANCE, totalFee);
+            }
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java:142]: If these exceptions can be the same, then we can simplify the throw blocks a bit by extracting the conditional.
At the beginning of the method, we can set the response code
```java
        final ResponseCodeEnum insufficientFundsResponse;
        if (ingestCheck) { // use a different status for ingest
            insufficientFundsResponse = INSUFFICIENT_PAYER_BALANCE;
        } else {
            insufficientFundsResponse = INSUFFICIENT_ACCOUNT_BALANCE;
        }
```
```suggestion
            throw new InsufficientNetworkFeeException(insufficientFundsResponse, totalFee);
```

Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java]: Yea, thought this code would be removed later and just did quick copy/paste. Updated now.

**Review by jsync-swirlds — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java [hunk 4]
```diff
@@ -124,17 +131,25 @@ public void checkSolvency(
...
-            throw new InsufficientNetworkFeeException(INSUFFICIENT_PAYER_BALANCE, totalFee);
+            throw new InsufficientNetworkFeeException(insufficientFeeResponseCode, totalFee);
         }
         if (offeredFee < totalFee) {
             throw new InsufficientServiceFeeException(INSUFFICIENT_TX_FEE, totalFee);
         }
+
         if (availableBalance < totalFee) {
-            throw new InsufficientServiceFeeException(INSUFFICIENT_PAYER_BALANCE, totalFee);
+
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/SolvencyPreCheck.java:151]: Extra newline here, but nbd

**Review by mhess-swl — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**